packages/components/src/mermaid.tsx

```diff
@@ -1,29 +1,68 @@
-import { ReactElement, useEffect, useState } from 'react';
-import mermaid, { Config } from 'mermaid';
+import { ReactElement, useEffect, useId, useState } from 'react';
+import mermaid from 'mermaid';
 import { useTheme } from 'nextra-theme-docs';
 
-/**
- * Assign a unique ID to each mermaid svg as per requirements of `mermaid.render`.
- */
-let id = 0;
+let initialized = false;
 
 export const Mermaid = ({ chart }: { chart: string }): ReactElement => {
   const { resolvedTheme } = useTheme();
-  const [svg, setSVG] = useState('');
+  const id = useId();
+  const [svg, setSvg] = useState('');
 
+  // must use useEffect bacause mermaid needs the browser
   useEffect(() => {
-    mermaid.initialize({
-      startOnLoad: true,
-      theme: resolvedTheme as Config['theme'],
-      fontFamily: 'inherit',
-    });
-    try {
-      const renderedSvg = mermaid.render(`mermaid-svg-${(id += 1)}`, chart);
-      setSVG(renderedSvg);
-    } catch (error) {
-      console.error('Error while rendering mermaid', error);
```

Comment:
Please keep console error, otherwise error can break rendering app https://github.com/vercel/next.js/issues/40750

Reply:
Ah ok, understood - https://github.com/the-guild-org/the-guild-components/pull/517/commits/d0c4440f08e5d50aae3cbae16207f7969db89f5d.